attending/library.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+
+from .downloader import write_to_file
+from .Doc import DocLocation, Doc
+
+
+def _module_identifier(module):
+    return module.__name__, module.__version__
+
+
+class Library:
+    def __init__(self, home=Path().home()):
+        self.location = self._construct(home / Path(".attending"))
+        self.docs = self._load_docs()
+
+    def fetch(self, module):
+        if not hasattr(module, "__doc_url__"):
+            raise AttributeError(f"{module.__name__} missing '__doc_url__'")
+        if not hasattr(module, "__version__"):
+            raise AttributeError(f"{module.__name__} missing '__version__'")
+        if module not in self:
+            doc_location = DocLocation(self.location, module.__name__, module.__version__)
+            self._add_project(doc_location, module.__doc_url__)
+
+    def _construct(self, home: Path):
+        if not home.exists():
+            home.mkdir(parents=True)
+        return home
+
+    def _add_project(self, doc_location: DocLocation, url):
+        if doc_location.full_path().exists():
+            raise FileExistsError(f"{doc_location.full_path()}")
+        doc_location.full_path().mkdir(parents=True)
+        write_to_file(url, doc_location)
+        self.docs[doc_location.as_module_identifier()] = Doc(doc_location.name, doc_location)
+
+    def _load_docs(self):
+        docs = {}
+        for project in self.location.iterdir():
+            if project.is_dir():
+                for version in project.iterdir():
+                    if version.is_dir():
+                        docs[(project.stem, version.name)] = \
+                            Doc(project.stem, DocLocation(self.location, project.stem, version.name))
+        return docs
+
+    def retire(self, module):
```

Comment:
this will remove a module from attending (and clean it up from the drive). Need to test this with multiple versions of the same module. Also, it leaves around a directory with the module name. This is fine unless it is the last module to be removed.

Reply:
cool. Your comment sounds like a good rough draft for the docstring.